### ISSUE TITLE
parse array bug

### ISSUE THREAD

**Issue by gulaandrij** [nestjs/nest#6263]
## Current behavior
when I try to validate post body with an array of elements it shows only validation errors for the first element only
![image](https://user-images.githubusercontent.com/2658746/106138599-0487ab00-6175-11eb-8565-64ca8d010b7f.png)
![image](https://user-images.githubusercontent.com/2658746/106138689-21bc7980-6175-11eb-9b34-6eeca0b9b1ae.png)

## Input Code
https://github.com/gulaandrij/nest-parse-array-pipe-bug

```curl
curl -X POST "http://localhost:3000/" -H  "accept: application/json" -H  "Content-Type: application/json" -d "[{\"number\":\"asd\",\"string\":\"string\",\"date\":\"2021-01-28T11:02:48.665Z\"},{\"number\":0,\"string\":3,\"date\":\"2021-01-28T11:02:48.665Z\"},{\"number\":0,\"string\":\"sdf\",\"date\":\"2021-22-28T11:02:48.665Z\"}]"
```

## Expected behavior
show errors for all array elements

## Environment

<pre><code>
Nest version: 7.6.7

 
For Tooling issues:
- Node version: 14
- Platform:  Mac

Others:

</code></pre>

**Comment by gulaandrij** [comment]
https://github.com/gulaandrij/nest-parse-array-pipe-bug/blob/master/src/app.controller.ts#L16
seems like this parameter not working

**Comment by kamilmysliwiec** [comment]
So actually, the `ParseArrayPipe` was supposed to stop at the first validation error, but maybe we should change that to avoid confusion with this `stopAtFirstError` configuration property you've used. Basically, this `stopAtFirstError` prop is inherited from the `class-validator` interface and isn't being used by the actual pipe at all. We can align the current behavior though and, in case "stopAtFirstError" is explicitly set to `false` (as in your example), we could validate all items in the array. Would you like to create a PR for this issue?

**Comment by gulaandrij** [comment]
I'm not sure how to fix this properly. I need to validate all array elements and receive all errors for all elements in response, not only for first one

**Comment by kamilmysliwiec** [comment]
Fixed in 7.6.9

**Comment by gulaandrij** [comment]
If we got several errors for one object response look like this:
![image](https://user-images.githubusercontent.com/2658746/106627934-f32e0c80-6581-11eb-8680-68864558efaa.png)

**Comment by gulaandrij** [comment]
@kamilmysliwiec ^

**Comment by jmcdo29** [comment]
What would you expect instead?

**Comment by gulaandrij** [comment]
smth like:
```json
{
  "statusCode": 400,
  "messages": [{
    "key": "some_key",
    "validations": ["error1", "error2"]
  }],
  "error": "Bad Request"
}
```

this will allow showing errors for some fields on the frontend

**Comment by jmcdo29** [comment]
Do you want to make a PR to put it in that format? You could probably modify the `exceptionsFactory` and pass a custom factory to make the output look like that actually.

**Comment by kamilmysliwiec** [comment]
Fixed in 6.9.10.

The default (expected) exception message for nested objects:

```
'[0] random.title must be a string',
'[1] random.isEnabled should not be null or undefined',
'[1] random.isEnabled must be a boolean value',
'[1] random.createdAt must be a Date instance',
```

where [i] is an index in the input array. And for nested arrays:

```
'[0] random.0.title must be a string',
'[0] random.1.title must be a string',
'[1] random.0.isEnabled should not be null or undefined',
'[1] random.0.isEnabled must be a boolean value',
'[1] random.0.createdAt must be a Date instance',
```

if you need more control, you can always modify the `exceptionFactory` as @jmcdo29 suggested above.
